integrations/aws/utils/resources.py

```diff
@@ -0,0 +1,180 @@
+    return service in global_services
+
+
+def fix_unserializable_date_properties(obj: Any) -> Any:
+    return json.loads(json.dumps(obj, default=str))
+
+
+async def describe_single_resource(
+    kind: str, identifier: str, account_id: str | None = None, region: str | None = None
+) -> dict[str, str]:
+    async for session in get_sessions(account_id, region):
+        region = session.region_name
+        try:
+            match kind:
+                case ResourceKindsWithSpecialHandling.ACM:
+                    async with session.client("acm") as acm:
+                        response = await acm.describe_certificate(
+                            CertificateArn=identifier
+                        )
+                        return response.get("Certificate")
+
+                case ResourceKindsWithSpecialHandling.LOADBALANCER:
+                    async with session.client("elbv2") as elbv2:
+                        response = await elbv2.describe_load_balancers(
+                            LoadBalancerArns=[identifier]
+                        )
+                        return response.get("LoadBalancers")[0]
+
+                case ResourceKindsWithSpecialHandling.CLOUDFORMATION:
+                    async with session.client("cloudformation") as cloudformation:
+                        response = await cloudformation.describe_stacks(
+                            StackName=identifier
+                        )
+                        return response.get("Stacks")[0]
+
+                case ResourceKindsWithSpecialHandling.EC2:
+                    async with session.client("ec2") as ec2_client:
+                        described_instance = await ec2_client.describe_instances(
+                            InstanceIds=[identifier]
+                        )
+                        instance_definition = described_instance["Reservations"][0][
+                            "Instances"
+                        ][0]
+                        return instance_definition
+
+                case _:
+                    async with session.client("cloudcontrol") as cloudcontrol:
+                        response = await cloudcontrol.get_resource(
+                            TypeName=kind, Identifier=identifier
+                        )
+                        resource_description = response.get("ResourceDescription")
+                        return {
+                            "Identifier": resource_description.get("Identifier"),
+                            **json.loads(resource_description.get("Properties", {})),
+                        }
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "ResourceNotFoundException":
+                logger.info(f"Resource not found: {kind} {identifier}")
+                return {}
+            logger.error(
+                f"Failed to describe CloudControl Instance in region: {region}; error {e}"
+            )
+            return {}
+    return {}
```

Comment:
instead of returning {} raise the error and handle it outside

Reply:
The last return is for the linter